examples/shader_advanced/custom_mesh_pass.rs

```diff
@@ -0,0 +1,266 @@
+        MeshPassPlugin::<OutlinePass>::default(),
+    ))
+    // You can use `register_required_components` to add our `OutlinePass` to all cameras.
+    // Example: .register_required_components::<Camera3d, OutlinePass>()
+    .add_systems(Startup, setup);
+
+    let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+        return;
+    };
+    render_app
+        .add_render_graph_node::<ViewNodeRunner<OutlinePassNode>>(Core3d, OutlinePassLabel)
+        .add_render_graph_edges(
+            Core3d,
+            (
+                Node3d::MainOpaquePass,
+                OutlinePassLabel,
+                Node3d::MainTransmissivePass,
+            ),
+        );
+
+    app.run();
+}
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone, RenderLabel)]
+struct OutlinePassLabel;
+
+#[derive(Clone, Copy, Default, Component, ExtractComponent)]
+struct OutlinePass;
+
+impl MeshPass for OutlinePass {
+    type ViewKeySource = MainPass;
+    type Specializer = MaterialPipelineSpecializer;
+    type PhaseItems = OutlineOpaque3d;
+    type RenderCommand = DrawMaterial;
+}
+
+#[derive(BinnedPhaseItem)]
+struct OutlineOpaque3d(Opaque3d);
+
+impl PhaseItemExt for OutlineOpaque3d {
+    type PhaseFamily = BinnedPhaseFamily<Self>;
+    type ExtractCondition = NoExtractCondition;
+    const PHASE_TYPES: RenderPhaseType = RenderPhaseType::Opaque;
+
+    fn queue(render_phase: &mut PIEPhase<Self>, context: &PhaseContext) {
+        let (vertex_slab, index_slab) = context
+            .mesh_allocator
+            .mesh_slabs(&context.mesh_instance.mesh_asset_id);
+        
+        render_phase.add(
+            Opaque3dBatchSetKey {
+                pipeline: context.pipeline_id,
+                draw_function: context.draw_function,
+                material_bind_group_index: Some(context.material.binding.group.0),
+                vertex_slab: vertex_slab.unwrap_or_default(),
+                index_slab,
+                lightmap_slab: context
+                    .mesh_instance
+                    .shared
+                    .lightmap_slab_index
+                    .map(|index| *index),
+            },
+            Opaque3dBinKey {
+                asset_id: context.mesh_instance.mesh_asset_id.into(),
+            },
+            (context.entity, context.main_entity),
+            context.mesh_instance.current_uniform_index,
+            BinnedRenderPhaseType::mesh(
+                context.mesh_instance.should_batch(),
+                &context.gpu_preprocessing_support,
+            ),
+            context.current_change_tick,
+        );
+    }
+}
+
+#[derive(Asset, TypePath, AsBindGroup, Clone, Default)]
+struct OutlineExtention {
+    #[uniform(100)]
+    outline_color: LinearRgba,
+}
+
+impl MaterialExtension for OutlineExtention {
+    fn shaders() -> PassShaders {
+        let mut pass_shaders = PassShaders::default();
+        pass_shaders.extend([
+            (MainPass::id(), ShaderSet::default()),
```

Comment:
Is this required for this example? I'm not sure why this entity would even need to run in the main pass?